server/routes/journeys/base/patchInvestigationController.ts

```diff
@@ -56,4 +56,66 @@ export class PatchInvestigationController extends BaseJourneyController {
     req.flash(FLASH_KEY__CSIP_SUCCESS_MESSAGE, successMessage)
     next()
   }
+
+  addInterview = async <T>({
+    req,
+    res,
+    next,
+    body,
+  }: {
+    req: Request<Record<string, string>, unknown, T>
+    res: Response
+    next: NextFunction
+    body: components['schemas']['CreateInterviewRequest']
+  }) => {
+    try {
+      await this.csipApiService.addInterview(req as Request, body)
+    } catch (e) {
+      if ((e as SanitisedError).data) {
+        const errorRespData = (e as SanitisedError).data as Record<string, string | unknown>
+        req.flash(
+          FLASH_KEY__VALIDATION_ERRORS,
+          JSON.stringify({
+            referral: [errorRespData?.['userMessage'] as string],
+          }),
+        )
+      }
+      res.redirect('back')
+      return
```

Comment:
Can we do some extracting on these API call methods? There's a lot of copied code doing the same thing in this file now

Reply:
Agreed :)